jax_galsim/wcs.py

```diff
@@ -0,0 +1,922 @@
+import jax.numpy as jnp
+
+from jax_galsim.gsobject import GSObject
+from jax_galsim.position import Position, PositionI, PositionD
+from jax_galsim.transform import _Transform
+
+import galsim as _galsim
+from jax._src.numpy.util import _wraps
+from jax.tree_util import register_pytree_node_class
+
+# We inherit from the reference BaseWCS and only redefine the methods that
+# make references to jax_galsim objects.
+class BaseWCS(_galsim.BaseWCS):
+    @_wraps(_galsim.BaseWCS.toWorld)
+    def toWorld(self, *args, **kwargs):
+        if len(args) == 1:
+            if isinstance(args[0], GSObject):
+                return self.profileToWorld(*args, **kwargs)
+            else:
+                return self.posToWorld(*args, **kwargs)
```

Comment:
I think it's because `profileToWorld` does not explicitly use one of the new jax objects, like `posToWorld` which uses the new object `Position`.

Reply:
Hi @b-remy - my bad, I think I was thinking of something else when I wrote this. I do agree with your confusion now. I think we should also redefine `profileToWorld` since it would create a `PositionD` jax-galsim object.

Reply:
why not reimplement `affine` that uses `_PositionD` (which we didn't reimplement so would use `PositoinD`)?